tests/scripts/build-release.sh

```diff
@@ -1,15 +1,34 @@
 #!/usr/bin/env bash
 set -ex
 
-#############
-# FUNCTIONS #
-#############
```

Comment:
Things are kind of scattered now with some behavior before the function definitions and some after.

What about leaving this header and creating a new MAIN header after the functions, then `moving the SHOULD_PROMOTE if/else under MAIN?